src/legacy/core_plugins/console/public/np_ready/lib/local_storage_object_client/local_storage_object_client.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import uuid from 'uuid';
+import { ObjectStorage, IdObject } from '../../../../common/types';
+import { Storage } from '../../services';
+
+export class LocalObjectStorage<O extends IdObject> implements ObjectStorage<O> {
+  private readonly prefix: string;
+
+  constructor(private readonly client: Storage, type: string) {
+    this.prefix = `console_local_${type}`;
+  }
+
+  async create(obj: Omit<O, 'id'>): Promise<O> {
+    const id = uuid.v4();
+    const newObj = { id, ...obj } as O;
```

Comment:
Does it work if you just specify the type of `newObj`? I usually try not to use casts like this unless I have to.

```suggestion
    const newObj: O = { id, ...obj };
```

Reply:
Good point!

Reply:
Ah, I see what the issue was here:

```
TS2322: Type '{ id: string; } & Pick<O, Exclude<keyof O, "id">>' is not assignable to type 'O'.   '{ id: string; } & Pick<O, Exclude<keyof O, "id">>' is assignable to the constraint of type 'O', but 'O' could be instantiated with a different subtype of constraint 'IdObject'.
```
So I just `as`'d it. Should be fairly safe here I think.